src/v/storage/kvstore.cc

```diff
@@ -78,7 +78,7 @@ ss::future<> kvstore::start() {
           _started = true;
 
           // Flushing background fiber
-          (void)ss::with_gate(_gate, [this] {
+          ssx::spawn_with_gate(_gate, [this] {
```

Comment:
optional: I wonder if it's worth simplifying the gate check:
```
                        if (_gate.is_closed()) {
                            return ss::now();
                        }
```
to
```
                        _gate.check();
```

Reply:
I'm kind of neutral on it -- it would work to use a `check()` but sometimes it's nice to explicitly handle expected cases with an `if` rather than an exception :shrug: